env-hooks/99.ed_object_models.bash

```diff
@@ -1,36 +1,34 @@
 function ed_object_models_setup {
-        local model_path=`rospack find ed_object_models`/models
-    
-    if [ -z $GAZEBO_MODEL_PATH ]
-    then
-        export GAZEBO_MODEL_PATH=$model_path
-    else
-        export GAZEBO_MODEL_PATH=$model_path:$GAZEBO_MODEL_PATH
-    fi
-    if [ -z $GAZEBO_RESOURCE_PATH ]
-    then
-        export GAZEBO_RESOURCE_PATH=$model_path
-    else
-        export GAZEBO_RESOURCE_PATH=$model_path:$GAZEBO_RESOURCE_PATH
-    fi
+    local model_path=$(rospack find ed_object_models)/models
 
     local model_paths=($(find $model_path -type f -name '*model*.sdf' | xargs dirname | xargs dirname))
-    local unique_model_paths=$(printf "%s\n" "${model_paths[@]}" | sort -u | tr '\n' ' ')
+    local unique_model_paths=$(printf "%s\n" "${model_paths[@]}" | sort -ru | tr '\n' ' ')
     local model_paths_string=""
     for dir in $unique_model_paths
     do
         if [[ "$dir" == "$model_path" ]]
         then
+            # You want to have the main model path at the end
             continue
         fi
-        model_paths_string="$dir:$model_paths_string"
+        model_paths_string="$model_paths_string:$dir"
```

Comment:
Do it this way:
`model_paths_string=${model_paths_string:+${model_paths_string}:$dir}`

Reply:
I don't know what it is supposed to do, but it gives me empty strings.

Reply:
Its a type of Parameter expansion. Check this:
https://unix.stackexchange.com/a/102899

It shouldn't give you empty strings. I have tested this formulation. `$dir` should get appended to the list. In the installation of CUDA, this formulation is used to add bin to PATH and set LD_LIBRARY